app/jobs/event/participation_confirmation_job.rb

```diff
@@ -9,28 +9,33 @@ class Event::ParticipationConfirmationJob < BaseJob
 
   self.parameters = [:participation_id, :locale]
 
-  def initialize(participation)
+  def initialize(participation, send_approval: true)
     super()
     @participation_id = participation.id
+    @send_approval = send_approval
   end
 
   def perform
     return unless participation # may have been deleted again
 
     set_locale
     send_confirmation
-    send_approval
+    send_approval if participation.pending?
```

Comment:
Ich bin etwas verwirrt von dem hier. Der Call selbst wird mittels `if` gesteuert und in der Methode selbst wird auch noch returned je nach Instanzvariable
Kann man das irgendwie zusammenfassen?

Reply:
glaube das ist schon korrekt so. wir senden die confirmation wenn die participation aktiv wird oder zurück auf pending wechselt. send_approval soll aber nur passieren wenn die participation noch nicht aktiv ist. 
der participation confirmation job wird ja auch im participations controller getriggert beim erstellen einer neuen participation